crates/harness-tests/src/routing.rs

```diff
@@ -322,3 +326,28 @@ async fn should_prioritize_same_region_ws(ctx: &mut TestRunner) {
     assert_eq!(recvd.len(), 1);
     assert!(matches!(recvd[0], MsgB2M::Join(_)));
 }
+
+#[test_context(TestRunner)]
+#[tokio::test]
+#[allow(dead_code)]
+async fn test_malformed_header_rsv2_rsv3(ctx: &mut TestRunner) {
+    let client = tokio_tungstenite::connect_async(ctx.url("ws", "/api/room/test")).await.expect("failed to connect");
+
+    let dataframe = Frame {
+        header: FrameHeader {    
+            is_final: true,
+            rsv1: false,
+            rsv2: true,
+            rsv3: true,
+            opcode: OpCode::Data(()),
```

Comment:
This should be text, not data.

Reply:
@dyc3 clarification, text as in: OpCode::Text or OpCode::Data("")?